src/docs/reference/modules/Media.AmazonS3/README.md

```diff
@@ -0,0 +1,136 @@
+# Amazon S3 Media Storage (`OrchardCore.Media.AmazonS3`)
+
+The Amazon Media Storage feature enables support for storing assets in Amazon S3 Bucket.
+
+The feature replaces the default App_Data file-based media store with an Amazon Media Storage Provider.
+
+Media is still served by the Orchard Core website, and the Media Cache module takes responsibility for fetching media, on the fly, from the Amazon S3 Bucket.
+
+This allows the Amazon Media Storage feature to support image resizing on the fly through integration with `ImageSharp.Web`.
+
+The URL generated by the AssetUrl helpers points to the Orchard Core website.
+
+## Configuration
+
+The following configuration values are used by default and can be customized:
+
+```json
+{
+   "OrchardCore": {
+       "OrchardCore_Media_Amazon_S3": {
+           // Your AWS S3 Bucket name
+           "BucketName": "",
+           // This section needed only if Orchard will be hosted not in the AWS Cloud
+           // You can obtain all that information in the IAM Management Console
+            "Credentials": {
+              "SecretKey": "",
+              "AccessKeyId": "",
+              "RegionEndpoint": ""
+           },
+           // Optionally, set to a path to store media in a subdirectory inside your container.
+           "BasePath": "/media",
+           // If you have aws cli installed and configured you may just specify profile name
+           "ProfileName": ""
+       }
+  }
+}
+```
+
+Refer also to the [Configuration Section](../../core/Configuration/README.md),
+and the [Media Section](../Media/README.md) for other Media related configuration settings.
+
+There are two hosting options: inside and outside AWS Cloud.
+In case you are hosting Orchard Core inside AWS (EC2, EKS, etc.) you need to configure only `BucketName` and you may delete or comment out the other sections.
+
+In case you are hosting Orchard Core outside of AWS, you should fill the `Credentials` section or if you have AWS CLI installed and configured on your server you may specify only configured profile name (`default` if a profile name was not chosen during AWS CLI configuration).
+
+You can find region endpoints in the [Official AWS S3 Documentation](https://docs.aws.amazon.com/general/latest/gr/s3.html), see Region column. For example for Frankfurt region you should use `eu-central-1` 
+
+## AWS S3 Bucket Configuration
+
+By default, AWS 3S Bucket has limitations for newly uploaded files. If you want media files to be available from the outside of AWS, you should set up your bucket permissions.
+
+The simplest way of doing it is to add a policy:
```

Comment:
```suggestion
Refer also to the [Configuration Section](../../core/Configuration/README.md),
and the [Media Section](../Media/README.md) for other Media related configuration settings.

There are two hosting options: inside and outside AWS Cloud.
In case you are hosting Orchard Core inside AWS (EC2, EKS, etc.) you need to configure only `BucketName` and you may delete or comment out the other sections.

In case you are hosting Orchard Core outside of AWS, you should fill the `Credentials` section or if you have AWS CLI installed and configured on your server you may specify only configured profile name (`default` if a profile name was not chosen during AWS CLI configuration).

You can find region endpoints in the [Official AWS S3 Documentation](https://docs.aws.amazon.com/general/latest/gr/s3.html), see Region column. For example for the Frankfurt region you should use `eu-central-1` 

## AWS S3 Bucket Configuration

By default, AWS 3S Bucket has limitations for newly uploaded files. If you want media files to be available from the outside of AWS, you should set up your bucket permissions.

The simplest way of doing it is to add a policy:
```